app/controllers/homeworks_controller.rb

```diff
@@ -14,7 +17,9 @@ def create
       create_jobs_and_tasks
       redirect_to @assignment.term.course
     else
-      render "new"
+      params[:id] = @assignment.term.course_id
+      params[:term_number] = @assignment.term.number
+      render 'new'
```

Comment:
This render change address from new_homework_path to "/homeworks". Is there a way to stay on new_homework_path and show errors?